### PR TITLE
add IsOneShot function to fit PodKill and AwsRestart

### PR DESCRIPTION
**Pull request by YangKeao**
Signed-off-by: YangKeao <keao.yang@yahoo.com>

1. Avoid pausing one-shot chaos. All one-shot chaos will keep "RunningPhase" after injection
2. Modify `IsChaosFinished` function and add test to fix bugs about confusing one-shot chaos and chaos without duration.

### CODE REVIEW

**Review by fewdan — APPROVED** [review]
LGTM

File: api/v1alpha1/podchaos_types.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ import (
 
 // +kubebuilder:object:root=true
 // +chaos-mesh:base
+// +chaos-mesh:oneshot=.Spec.Action==PodKillAction
```
Comment (cwen0) [inline api/v1alpha1/podchaos_types.go:22]: `ContainerKill`  also belongs to `oneshot`?

Comment (YangKeao) [reply api/v1alpha1/podchaos_types.go]: Yes! Fixed.

**Review by cwen0 — APPROVED** [review]
LGTM

**Comment by cwen0** [comment]
@YangKeao e2e test was broken!

**Comment by STRRL** [comment]
@YangKeao e2e test case need update: https://github.com/chaos-mesh/chaos-mesh/blob/c8be4ad34e328728d42b2d39766d7c3e63107c95/e2e-test/e2e/chaos/podchaos/pod_kill.go#L144-L153

We still have a test case with "pause" on that one-shot chaos

**Pull request merged**